### DISCUSSION TITLE
请教用cmd调用qrcode_read时中值滤波怎么传参数

### DISCUSSION THREAD

**Discussion by cstybbt** [hiroi-sora/Umi-OCR#474] [Q&A]
在软件上把中值滤波改到7后能识别的二维码在cmd中不能识别出来。
看到是.settings里面已经设置了preprocessing.median_filter_size=7
那么cmd中怎么把这个参数传递进去？

**Comment by hiroi-sora** [comment]
你好，命令行调用的二维码识别 不与软件界面的设置挂钩，是独立的设置。

目前并没有设计传参指令，需要修改代码实现修改命令行设置。

你可以打开 `UmiOCR-data\py_src\server\cmd_server.py` ，第291行为：
```python
                res = MissionQRCode.addMissionWait({}, [{"pil": pil}])
```
将这一行改为：
```python
                argd = {
                    "preprocessing.median_filter_size": 7,
                }
                res = MissionQRCode.addMissionWait(argd, [{"pil": pil}])
```
同理，你还想添加什么参数，就写进 argd 字典中。

**Reply by cstybbt** [reply]
好的 谢谢！
